lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -475,13 +475,23 @@ def record_apply_failure(err, resources: [])
         .select(&:sensitive_template_content?)
         .map { |r| File.basename(r.file_path) }
 
+      server_dry_run_validated_resource = resources
+        .select(&:server_dry_run_validated?)
+        .map { |r| File.basename(r.file_path) }
+
       err.each_line do |line|
         bad_files = find_bad_files_from_kubectl_output(line)
         if bad_files.present?
```

Comment:
This conditional is getting very nested at this point. Maybe it would be clearer if we turn this piece into an `unless` (and used `next` in it), to at least get rid of one layer?

Reply:
I update the nested if logic here, easy to check out the change 🙂 : https://github.com/Shopify/kubernetes-deploy/pull/570/commits/ff7bbb1e32e7970d62eeae14aa3c7418417aacf0